specification/basic_concepts/basic_concepts.tex

```diff
@@ -1,33 +1,62 @@
 \chapter{Basic concepts}\label{sec:basic_concepts}
 
+\section{Main principles}
 
 A UAVCAN network is a decentralized peer network, where each peer (node) has a unique
-numeric identifier --- \emph{node ID} --- ranging from 1 to 127, inclusively.
+numeric identifier\footnote{Here and elsewhere in this specification, \emph{ID} and \emph{identifier} are used
+interchangeably unless specifically indicated otherwise.}
+--- \emph{node ID} --- ranging from 1 to 127, inclusively.
 Nodes of a UAVCAN network can communicate using the following communication methods:
 
 \begin{description}
-    \item[Message broadcasting] --- The primary method of data exchange with one-to-all publish/subscribe semantics.
-    \item[Service invocation] --- The communication method for peer-to-peer request/response
+    \item[Message publication] --- The primary method of data exchange with one-to-many publish/subscribe semantics.
+    \item[Service invocation] --- The communication method for one-to-one request/response
     interactions\footnote{Like remote procedure call (RPC).}.
 \end{description}
 
-For each type of communication, a predefined set of data structures is used,
-where each data structure has a unique identifier --- the \emph{data type ID} (DTID).
-Additionally, every data structure definition has a pair of major and minor semantic version numbers,
+For each type of communication, a predefined set of data types is used,
+where each data type has a unique name.
+Additionally, every data type definition has a pair of major and minor version numbers,
 which enable data type definitions to evolve in arbitrary ways while ensuring a well-defined
 migration path if backward-incompatible changes are introduced.
-Some data structures are standard and defined by the protocol specification (of which only a small 
+Some data types are standard and defined by the protocol specification (of which only a small
```

Comment:
What's being said here is that some types are defined by the "protocol specification", not just "protocol". We could rephrase it to state "defined by the specification" if that adds clarity. I don't think we can just throw in a word "library" without explaining what it is exactly (this term does not appear anywhere in the specification currently, we only introduce "root namespaces" as the closest thing to a library).

Reply:
I see the problem. What I'm worried about is that the sentence may mislead readers to expect to find the standard types defined inside the protocol specification document. I'm happy with any rephrasing that will make the core protocol a bit more seperated from the standard dsdl.

Perhaps it would work out to say "There exists standard types that defines....." leaving the explaining of how/where they are defined till later.

Reply:
> What I'm worried about is that the sentence may mislead readers to expect to find the standard types defined inside the protocol specification document.

Ooh, I was actually considering to do just that, but still not sure if it makes sense. We could fully automate the task, too. This should be discussed when we get closer to Chapter 5 "Application layer".

Reply:
I would avoid doing this for any other types than the most essential:
 - NodeStatus/Heartbeat is required
 - GetNodeInfo is nice to do it for
 - GetPortInfo (and friends?) is nice to do it for

I think doing it for file, diagnostics, etc would be a bit messy since directory structure (namespacing) is not preserved well in a spec sheet.